FWIO/RNTuple/plugins/RNTupleOutputFile.cc

```
@@ -0,0 +1,583 @@
+      return iName.substr(0, iName.size() - 1);
+    }
+
+    void noSplitField(ROOT::Experimental::RFieldBase& iField) {
+      auto const& typeName = iField.GetTypeName();
+      if (typeName == "std::uint16_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kUInt16}});
+      } else if (typeName == "std::uint32_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kUInt32}});
+      } else if (typeName == "std::uint64_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kUInt64}});
+      } else if (typeName == "std::int16_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kInt16}});
+      } else if (typeName == "std::int32_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kInt32}});
+      } else if (typeName == "std::int64_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kInt64}});
+      } else if (typeName == "float") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kReal32}});
+      } else if (typeName == "double") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kReal64}});
+      }
+    }
+
+    void checkSubFieldsForNoSplit(ROOT::Experimental::RFieldBase& iField,
+                                  std::vector<std::string> const& iNoSplitFields) {
+      for (auto const& name : iNoSplitFields) {
+        if (name.starts_with(iField.GetFieldName())) {
+          bool found = false;
+          for (auto& subfield : iField) {
+            if (subfield.GetQualifiedFieldName() == name) {
+              found = true;
+              noSplitField(subfield);
+              break;
+            }
+          }
+          if (not found) {
+            edm::LogError("MissingSubField")
+                << "The data product was found but the requested subfield '" << name << "' is not part of the class";
```

Should we throw an exception instead? If not, some explanation of why not would be nice (in comments).

---

Throwing probably makes the most sense.